addons/account/models/account_move.py

```diff
@@ -1793,6 +1794,75 @@ def _build_credit_warning_message(self, record, updated_credit):
         msg += ': %s' % formatLang(self.env, updated_credit, currency_obj=record.company_id.currency_id)
         return msg
 
+    @api.depends('ref', 'move_type', 'partner_id', 'invoice_date')
+    def _compute_duplicated_ref_ids(self):
+        # The computed field only appears in the move view form and thus is supposed
+        # to only be computed for one move at a time which makes the search in loop "ok"
+        move_to_duplicate_move = self._fetch_duplicate_supplier_reference()
+        for move in self:
+            move.duplicated_ref_ids = move_to_duplicate_move.get(move, self.env['account.move'])
+
+
+    def _fetch_duplicate_supplier_reference(self, only_posted=False):
+        move_to_duplicate_moves = defaultdict(lambda x: self.env['account.move'])
+        moves = self.filtered(lambda m: m.is_purchase_document() and m.ref)
+        if not moves:
+            return move_to_duplicate_moves
+
+        # this case deals with a move created in UI having id being a NEW_ID object
+        if len(moves) == 1 and not moves.id:
+            duplicates = self.env["account.move"].search(
+                [
+                    ('company_id', '=', moves.company_id.id),
+                    ('move_type', '=', moves.move_type),
+                    ('commercial_partner_id', '=', moves.commercial_partner_id.id),
+                    ('ref', '=', moves.ref)
+                ]
+                + [
+                    ('invoice_date', '=', moves.invoice_date),
+                    ('state', '=', 'posted'),
+                ] if only_posted else []
+            )
+            move_to_duplicate_moves.update({
+                moves: duplicates
+            })
+            return move_to_duplicate_moves
```

Comment:
Not sure it will help.
I mainly put it for the case in which I don't have ids because it would requires me to inject the values in the query.
I could make it work that would probably be much longer.
As I am doing that only when there is only one record in creation through the UI, it's ok to directly go through the ORM.